07-routing-gossip.md

```diff
@@ -0,0 +1,113 @@
+This reflects the willingness of the announcing node to forward payments over the announced channel.
+
+## Message Format
+
+This specification defines one message type with identifier 256 (`0x00000100`), called a `MSG_ROUTING_UPDATE`.
+It comprises both the sending node's identity and contact information as well as all of its outgoing channels.
+
+```
+[4:timestamp]
+[33:node_id]
+[16:ipv6]
+[2:port]
+[3:rgb_color]
+[var:alias]
+[4+X*46:channels]
+[64:signature]
+```
+
+The timestamp is required to be monotonically increasing from one update to another, it MAY correspond to unix timestamp when the sending node created the announcement.
+The node ID is a 33 byte compressed public key that uniquely identifies the node in the network and matches the signature.
+The IPv6 field is used for both IPv4 and IPv6 addresses, using the IPv4-Mapped IPv6 Address format defined in [RFC 4291 section 2.5.5.2](https://tools.ietf.org/html/rfc4291#section-2.5.5.2):
+
+```
+|                80 bits               | 16 |      32 bits        |
++--------------------------------------+--------------------------+
+|0000..............................0000|FFFF|    IPv4 address     |
++--------------------------------------+----+---------------------+
+```
+
+The alias is a length prefixed UTF-8 encoded string.
+The UTF-8 encoded string is prefixed by a 4 byte unsigned int in network byte order, defining the length of the string.
+The RGB color and alias MAY be used by the node operator to customize their node's appearance in maps and graphs, and MAY be used to reference nodes in interfaces.
+Notice however that there is no collision protection for aliases, hence the node ID MUST be verified before initiating any transfer.
+
+The `channels` field comprises a 4 byte unsigned int count of channels that follow.
+Each channel has the following format:
+
+```
+[33:destination]
+[4:blockheight]
+[4:blockindex]
+[1:expiry]
+[2:fee_base]
+[2:fee_proportional]
+```
+
+The `destination` field is the node ID of the remote node from the point of view of the announcing node.
+Thus the `node_id` from the enclosing announcement and the `destination` identify the two endpoints of the channel.
+The `blockheight` and `blockindex` MAY be combined to form a globally unique `channel_id` that MAY be used to identify the channel in future.
+The remaining fields determine the minimum timelock difference that HTLCs have to respect (`expiry`) and the fees that the announcing node expected when routing a payment through the announced channel from `node_id` to `destination`.
+
+## Creation and Processing of Announcements
+
+The announcing node creates the message with the node's information and all its channels.
+Normal removal of a channel is done by omitting the channel in the `channels` field.
+Notice that this does not allow removing a channel if no active channels are left open, since an announcement requires at least one channel in the `channels` field to be valid.
```

Comment:
The problem is not so much accepting messages without channels the problem is that we'd have to forward them as well, flooding the network with messages that are basically free to create.

An attacker can just generate any number of node IDs and announce its existence. And we can't just drop empty announces for which we don't know the corresponding node, because other nodes may know it and require the removal message.

If we require a channel, even a closed one, he must at least make a bitcoin transaction, which makes the attack non-free. I agree that signaling via the `expiry` is a bit wonky, but keep in mind that this is voluntary signaling, we can still remove channels if the anchor TX gets spent (modulo effort of tracking those).